### ISSUE TITLE
Array values that are buffered by `.holdWhen()` are released item per item

### ISSUE THREAD

**Issue by timmolendijk** [baconjs/bacon.js#543]
I guess this must be a bug.

Compare:

```javascript
Bacon.interval(1000, [1,2]).
  take(3).log();

/* logs:
> [1, 2]
> [1, 2]
> [1, 2]
> <end>
*/
```

to

```javascript
Bacon.interval(1000, [1,2]).
  holdWhen(Bacon.later(1000, false).startWith(true)).
  take(3).log();

/* logs:
> 1
> 2
> [1, 2]
> <end>
*/
```

I've tried to [dig in the code](https://github.com/baconjs/bacon.js/blob/master/src/holdwhen.coffee) but I'm having a hard time understanding what's going on there.

**Comment by raimohanska** [comment]
Nice find! There is a bug indeed in the holdWhen code here: https://github.com/baconjs/bacon.js/blob/master/src/holdwhen.coffee#L19

The problem lies with the usage of `Array.prototype.concat` here: 

```coffeescript
(xs,x) -> xs.concat(x)
```

The concat function treats arrays differently from other values. Would you like to try and fix it?

**Comment by raimohanska** [comment]
Released 0.7.44 with fix
